adapter/internal/discovery/xds/server.go

```diff
@@ -270,14 +270,14 @@ func UpdateAPI(vHost string, apiProject mgw.ProjectAPI, environments []string) (
 		if err != nil {
 			return nil, err
 		}
-		//set swagger file configs not available
-		if mgwSwagger.GetSecurityScheme() == nil {
-			for _, value := range apiYaml.SecurityScheme {
-				schemes = append(schemes, model.SecurityScheme{Type: value})
+		schemes = mgwSwagger.GetSecurityScheme()
+		for _, value := range apiYaml.SecurityScheme {
+			if value == model.APIKeyInAppLevelSecurity {
+				schemes = append(schemes, model.SecurityScheme{DefinitionName: model.APIKeyInAppLevelSecurity,
+					Type: value, Name: model.APIKeyNameWithApim})
 			}
-			mgwSwagger.SetSecurityScheme(schemes)
 		}
-		mgwSwagger.SetXWso2AuthHeader(apiYaml.AuthorizationHeader)
```

Comment:
Add this back

Reply:
fixed with https://github.com/wso2/product-microgateway/pull/2432/commits/c96313affe6a7144b4c8daa0d2f7c1d35d61a2a9